cardano-testnet/src/Testnet/Util/Runtime.hs

```diff
@@ -99,6 +100,23 @@ data LeadershipSlot = LeadershipSlot
   , slotTime    :: Text
   } deriving (Eq, Show, Generic, FromJSON)
 
+-- Temporary path used at runtime
+newtype TmpAbsolutePath = TmpAbsolutePath
+  { unTmpPath :: FilePath
+  } deriving (Eq, Show)
+
+makeTmpRelPath :: TmpAbsolutePath -> FilePath
+makeTmpRelPath (TmpAbsolutePath fp) = makeRelative (makeTmpBaseAbsPath (TmpAbsolutePath fp)) fp
+
+makeSocketDir :: TmpAbsolutePath -> FilePath
+makeSocketDir fp = makeTmpRelPath fp </> "socket"
+
+makeTmpBaseAbsPath :: TmpAbsolutePath -> FilePath
+makeTmpBaseAbsPath (TmpAbsolutePath fp) = takeDirectory fp
```

Comment:
⬆️  parent directory of the temp absolute path.  The only reason we care about this directory is because [there is a limit to the length of the name of the socket](https://unix.stackexchange.com/questions/367008/why-is-socket-path-length-limited-to-a-hundred-chars).  Because temporary directories could be sufficiently long to blow this limit, we spawn processes like `cardano-node-chairman` in the `tmpBaseAbsPath` and use the filename of the socket relative to this location.  Relative socket filenames are not in danger of being too long even if the temporary directory is very long.

Reply:
Very good info. I'll add a comment in the code for that.
It is very difficult to make sense of the code without that info.

Reply:
Do you remember if that restriction only applies to Windows ?